Modules/Devices/src/Devices.Infrastructure/PushNotifications/IServiceCollectionExtensions.cs

```diff
@@ -21,20 +26,34 @@ public static void AddPushNotifications(this IServiceCollection services, PushNo
             case PROVIDER_DUMMY: 
                 services.AddDummyPushNotifications();
                 break;
+            case PROVIDER_DIRECT:
+                FirebaseApp.Create(new AppOptions
+                {
+                    Credential = options.DirectPnsCommunication is null
+                        ? GoogleCredential.GetApplicationDefault()
+                        : GoogleCredential.FromJson(options.DirectPnsCommunication.Fcm.ServiceAccountJson)
+                });
+
+                services.AddTransient<PnsConnectorFactory, PnsConnectorFactoryImpl>();
+                services.AddTransient<FirebaseCloudMessagingConnector>();
+                services.AddTransient<IPushService, DirectPushService>();
+                services.AddSingleton(FirebaseMessaging.DefaultInstance);
+                break;
```

Comment:
Extract all of this into a separate extension method, just like with AzureNotificationHub or the Dummy service